tests/ignite/handlers/test_lr_finder.py

```diff
@@ -321,6 +321,18 @@ def test_detach_terminates(lr_finder, to_save, dummy_engine, dataloader, recwarn
     # assert len(recwarn) == 0
 
 
+def test_different_num_iters(lr_finder, to_save, dummy_engine, dataloader):
+    with lr_finder.attach(dummy_engine, to_save, num_iter=200) as trainer_with_finder:
+        trainer_with_finder.run(dataloader)
+    len_lrs_few_iters = len(lr_finder._history["lr"])
+
+    with lr_finder.attach(dummy_engine, to_save, num_iter=1000) as trainer_with_finder:
+        trainer_with_finder.run(dataloader)
+    len_lrs_many_iters = len(lr_finder._history["lr"])
+
+    assert len_lrs_many_iters - len_lrs_few_iters > 500
```

Comment:
I think the test should ensure that trainer_with_finder.state.iteration after the run is equal to num_iter

Reply:
I tried that, the number of iterations always is not 100% equal to `num_iters`, it's always slightly smaller (that was always the case, even before these updates), how much smaller depends on the scheduler we are using, so If we would like to this, we should do range, for example
```py
num_iter = 100
num_iter - trainer_with_finder.state.iteration < 10
```

Reply:
That's true ! Maybe we can set the divergence threshold rather high such that we wont exit before ? In this case they should be exact. Or specify end_lr not that high such that loss value wont go too high neither ...